test/e2e/test/default.go

```diff
@@ -31,8 +38,12 @@ func DefaultSecurityContext() *corev1.PodSecurityContext {
 // with special permissions is created by APM test's builder
 // so that this can work.
 func APMDefaultSecurityContext() *corev1.PodSecurityContext {
-	defaultUserID := int64(1000)
-
+	defaultUserID := int64(12345) // arbitrary user ID
+	// APM image expected to run with the user ID 1000 or 0 before 7.9.0 (https://github.com/elastic/beats/issues/18871)
+	stackVersion := version.MustParse(Ctx().ElasticStackVersion)
+	if stackVersion.LT(version.MustParse("7.9.0")) {
```

Comment:
It works, but for versions > 7.9.0 we should ideally let OpenShift assigns the "UID" as it is done for the other Stack components with `DefaultSecurityContext()`:

```golang
	if !Ctx().OcpCluster {
		// if not in an OCP cluster manage the UID
	}
        // if in an OCP cluster, OpenShift sets the security context automatically
```

It would also mean that we can use the `default` `ServiceAccount` for the APM Server, instead of the one used to escape from the restricted SCC today: https://github.com/elastic/cloud-on-k8s/blob/f52ca4700f56238c56186f0fc751ada563cf6dda/test/e2e/test/apmserver/steps_creation.go#L32-L35

It might be a good opportunity to update this TODO.

That being, as I said, it works and you should have the pretty same result with your proposal.

Reply:
Yes, let's clean up.